Implementation/FindMyBLEDevice/FindMyBLEDevice/ViewModels/ItemsViewModel.cs

```diff
@@ -112,24 +128,27 @@ async void OnItemSelected(BTDevice device)
             await Shell.Current.GoToAsync($"{nameof(ItemDetailPage)}?{nameof(ItemDetailViewModel.DeviceId)}={device.Id}");
         }
 
-        async void SelectFoundDevice(AvailableBTDevice device)
+        async void OnAvailableDeviceSelected(AvailableBTDevice device)
         {
             if (device == null)
                 return;
 
             await App.Bluetooth.StopSearch();
 
-            // This will push the NewItemPage onto the navigation stack
-            await Shell.Current.GoToAsync($"{nameof(NewItemPage)}?{nameof(NewItemViewModel.BTGUID)}={device.Id}&{nameof(NewItemViewModel.AdvertisedName)}={device.Name}");
+            // This will replace the navigation stack with StrengthPage:
+            // TODO: find a way to just push the site onto the navigation stack
+            await Shell.Current.GoToAsync($"../StrengthPage?bt_id={device.Id}");
 
+            // Alternative: open add item page instead of signal strength page
+            // This will push the NewItemPage onto the navigation stack
+            //await Shell.Current.GoToAsync($"{nameof(NewItemPage)}?{nameof(NewItemViewModel.BTGUID)}={device.Id}&{nameof(NewItemViewModel.AdvertisedName)}={device.Name}");
         }
 
-
-        private async Task ExecuteSearchDevicesCommand()
+        private async Task ExecuteSearchAvailableDevicesCommand()
         {
             List<BTDevice> savedDevices = await App.DevicesStore.GetAllDevices();
-            await App.Bluetooth.Search(20000, AvailableDevices, found => savedDevices.Exists(saved => saved.BT_GUID.Equals(found.Id.ToString())));
```

Comment:
Is there any specific reason you removed that filter?

Reply:
I just un-did this change myself real quick.